<launch>
    
    <param name="scan_line" type="int" value="64" />

    <!-- if 1, do mapping 10 Hz, if 2, do mapping 5 Hz. Suggest to use 1, it will adjust frequence automatically -->
    <param name="mapping_skip_frame" type="int" value="1" />

    <!-- remove too closed points -->
    <param name="minimum_range" type="double" value="5"/>


    <param name="mapping_line_resolution" type="double" value="0.3"/>
    <param name="mapping_plane_resolution" type="double" value="0.6"/>
    <param name="curvature_threshold" type="double" value="0.1"/>
    <param name="point_to_point_weight" type="double" value="-0.5"/>
    <node pkg="aloam_velodyne" type="curvatureEstimate" name="curvatureEstimate" output="screen" />

    <!-- node pkg="aloam_velodyne" type="ascanRegistration" name="ascanRegistration" output="screen" -->
    <!-- node pkg="aloam_velodyne" type="alaserOdometry" name="alaserOdometry" output="screen" -->
    <!-- node pkg="aloam_velodyne" type="alaserMapping" name="alaserMapping" output="screen" -->
    <node pkg="aloam_velodyne" type="mapping" name="mapping" output="screen" />

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find aloam_velodyne)/rviz_cfg/rviz.rviz" />
        <!-- node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find aloam_velodyne)/rviz_cfg/comp.rviz" -->
    </group>

</launch>
